.github/workflows/ci.yml

```
@@ -189,6 +189,13 @@ jobs:
     - run: nox -e ${{ matrix.task.nox }}
       if: ${{ matrix.task.run-if }}
 
+  package:
+    name: Build & verify package
```

```suggestion
  package-validation:
    name: Verify package
```

I think that at this point, the build part is secondary.

We will not publish the package that is built in this job.
So having the "build" name, might suggest that this is the part where we build the package that will be published.

I know that most of the time, the package built here is the same as the one that is built in the "build" job (line 14)

```
jobs:
  build:
```

And maybe, we should not have a separate job.
Instead, just add this step to the existing `build` job.